app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,40 @@
+class TasksController < AuthenticatedController
+  include ProjectScoped
+
+  skip_before_action :set_project, if: -> { params[:project_id].blank? }
+  skip_before_action :set_nodes, if: -> { params[:project_id].blank? }
+
+  layout :set_layout
+  def index
+    @default_columns = ['Title', 'Methodology', 'Due Date', 'Assigned']
+
+    if params[:project_id].present?
+      cards = current_project.methodology_library.boards.map(&:cards)[0]
+      @local_storage_key = "project.ce.project_#{current_project.id}.tasks_datatable"
+    else
+      cards = current_user.cards
+      @default_columns.insert(1, 'Project')
+      @local_storage_key = 'project.pro.tasks_datatable'
+    end
+
+    @tasks = if cards
+      cards.filter_map { |card| card if card.assignees.include? current_user }
+    else
+      []
+    end
+  end
+
+  protected
+
+  # ProjectScoped always calls current_project. We are overwriting it here to
+  # prevent errors in Pro when viewing tasks outside of projects.
+  def current_project
```

Comment:
Instead of including it and opting out of several methods, what if we don't include `ProjectScoped` and call only the methods we need when we need them?

ex. Can we instead look up the project here using `params[:project_id]` if it's present?

Reply:
I tried that but we also need access to current_project in the views and ProjectScoped also sets `@nodes` for us (used by the main sidebar).

This is also the approach we take in https://github.com/dradis/dradis-ce/blob/develop/app/controllers/notifications_controller.rb